tensorflow_probability/python/experimental/distributions/laplace_approximation_test.py

```diff
@@ -0,0 +1,147 @@
+
+from tensorflow_probability.python.internal import test_util
+
+tfd = tfp.distributions
+tfb = tfp.bijectors
+tfde = tfp.experimental.distributions
+
+ATOL = 1e-4
+
+@test_util.test_graph_mode_only
+class LaplaceApproximationTest(test_util.TestCase):
+
+  def testJointDistributionCoroutineAutoBatchedWithDefaults(self):
+    @tfd.JointDistributionCoroutineAutoBatched
+    def joint_dist():
+      yield tfd.Normal(loc=1., scale=1.)
+      yield tfd.Normal(loc=[2., 2.], scale=2.)
+      yield tfd.Gamma(concentration=[3., 3., 3.], rate=10.)
+
+    approx = tfde.laplace_approximation(joint_dist)
+    mean = approx.bijector(approx.distribution.mean())
+    a, b, c = self.evaluate(mean)
+
+    self.assertEqual(len(mean), 3)
+
+    self.assertAllClose(a, 1., atol=ATOL)
+    self.assertAllClose(b, [2., 2.], atol=ATOL)
+    # this is the mode of the gamma distribution, not the mean
+    self.assertAllClose(c, [0.2, 0.2, 0.2], atol=ATOL)
+
+  def testJointDistributionCoroutineWithData(self):
+    Root = tfd.JointDistributionCoroutine.Root
+
+    @tfd.JointDistributionCoroutine
+    def joint_dist():
+      a = yield Root(tfd.Normal(loc=1., scale=1., name="a"))
+      yield tfd.MultivariateNormalDiag(loc=a + [2., 2.], scale_diag=[2., 2.])
+
+    approx = tfde.laplace_approximation(joint_dist, data={"a": 5.})
+    mean = self.evaluate(approx.distribution.mean())
+
+    self.assertAllClose(mean, [7., 7.], atol=ATOL)
+
+
+  def testJointDistributionNamedWithDefaults(self):
+    joint_dist = tfd.JointDistributionNamed(collections.OrderedDict(
+        a=tfd.Normal(loc=1.5, scale=2.),
+        b=lambda a: tfd.Normal(loc=a + 2.5, scale=3.),
+    ))
+
+    approx = tfde.laplace_approximation(joint_dist)
+    mvn = approx.distribution
+    mean, variance = self.evaluate([mvn.mean(), mvn.variance()])
+
+    self.assertAllClose(mean[0], 1.5, atol=ATOL)
+    self.assertAllClose(mean[1], 4., atol=ATOL)
+
+    self.assertAllClose(variance[0], 2.**2, atol=ATOL)
+    self.assertAllClose(variance[1], 2.**2 + 3.**2, atol=ATOL)
+
+
+  def testJointDistributionNamedWithBijector(self):
+    joint_dist = tfd.JointDistributionNamed(collections.OrderedDict(
+        a=tfd.Normal(loc=1.5, scale=2.),
+        b=tfd.Gamma(concentration=2.5, rate=3.),
+    ))
+
+    approx = tfde.laplace_approximation(
+        joint_dist, bijectors=[tfb.Identity(), tfb.Exp()])
+    mean = approx.distribution.mean()
+    a_mean, b_mean = self.evaluate(mean)
+
+    # the mean of the underlying unconstrained mvn
+    self.assertAllClose(a_mean, 1.5, atol=ATOL)
+    self.assertAllClose(b_mean, np.log((2.5 - 1.) / 3.), atol=ATOL)
+
+
+  def testJointDistributionSequentialAutoBatchedWithDefaults(self):
+    joint_dist = tfd.JointDistributionSequentialAutoBatched([
+        tfd.MultivariateNormalDiag(loc=0.123, scale_diag=[1., 2., 3.]),
+        tfd.StudentT(loc=2.5, scale=3., df=3.),
+    ])
+
+    approx = tfde.laplace_approximation(joint_dist)
+    mean = approx.bijector(approx.distribution.mean())
```

Comment:
why does this need to go through the bijector here? (I _think_ the bijector should be the identity here anyways, right? so maybe it doesn't...)

Reply:
here the bijector only splits up the MVN samples according to the event shape of the distributions in the joint distribution (`[3]` and `[]` in this case):

``` python
>>> approx.distribution.mean()
<tf.Tensor: shape=(4,), dtype=float32, numpy=array([0.12300005, 0.12300101, 0.12300146, 2.500002  ], dtype=float32)>
>>> approx.bijector(approx.distribution.mean())
_TupleWrapper((<tf.Tensor: shape=(3,), dtype=float32, numpy=array([0.12300005, 0.12300101, 0.12300146], dtype=float32)>, <tf.Tensor: shape=(), dtype=float32, numpy=2.500002>))
```

so have just used it for some convenience.